src/Sema.zig

```diff
@@ -17328,7 +17328,7 @@ fn zirPtrType(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!Air
         break :blk abi_align;
     } else 0;
 
-    const address_space = if (inst_data.flags.has_addrspace) blk: {
+    const address_space = if (elem_ty.zigTypeTag() == .Fn and target.cpu.arch == .avr) .flash else if (inst_data.flags.has_addrspace) blk: {
```

Comment:
This now silently overrides an explicitly specified addresspace.

Reply:
That's not good - updated now to preserve explicit address space.